packages/angular/projects/ui-angular/src/lib/primitives-poc/view/view-component/ComponentSetting.ts

```diff
@@ -0,0 +1,16 @@
+export class ComponentSettings {
```

Comment:
Is using a class necessary here? Other alternatives are (1) using a JSON object (which you can strictly type) and (2) exporting each constants out flatly

Reply:
Remove all the things in basecomponent.